### PR TITLE
[WIP] Plugin Updates

### PR DESCRIPTION
**Pull request by SchrodingersGat**
This PR introduces some new plugin enhancements which improve "ease of use" of plugin management:

### Features

- [x] UI updates
- [x] Extract installed package name for package plugins on registry load
- [x] Adds ability to update an existing plugin to the latest version (or to a particular version)
- [x] Improve error logging for plugin installation functions
- [x] Adds ability to specify version number when installing a plugin
- [x] Uninstall a deactivated plugin
- [x] Add ability to also delete `PluginConfig` when uninstalling a plugin
- [x] Improve update to `plugins.txt`
- [x] Remove entry from `plugins.txt` when uninstalling a plugin
- [x] Delete a `PluginConfig` associated with an uninstalled plugin

### Potential Future Updates
- [ ] Periodically check for available updates
 - [ ] Delete database tables associated with `app` plugin

### CODE REVIEW

**Comment by SchrodingersGat** [comment]
@matmair general thoughts on this one?

File: src/frontend/src/components/forms/ApiForm.tsx [hunk 1]
```diff
@@ -296,6 +297,7 @@ export function ApiForm({ id, props }: { id: string; props: ApiFormProps }) {
       method: method,
       url: url,
       data: data,
+      timeout: props.timeout ?? 5000,
```
Comment (matmair) [inline src/frontend/src/components/forms/ApiForm.tsx:300]: Comment: 5 seconds seam very long as a default

File: src/frontend/src/tables/part/PartTestTemplateTable.tsx [hunk 2]
```diff
@@ -85,9 +85,7 @@ export default function PartTestTemplateTable({ partId }: { partId: number }) {
     onFormSuccess: table.refreshTable
   });
 
-  const [selectedTest, setSelectedTest] = useState<number | undefined>(
-    undefined
-  );
+  const [selectedTest, setSelectedTest] = useState<number>(-1);
```
Comment (matmair) [inline src/frontend/src/tables/part/PartTestTemplateTable.tsx:88]: Question: Why is this in this diff?

File: InvenTree/plugin/installer.py [hunk 3]
```diff
@@ -195,30 +236,82 @@ def install_plugin(url=None, packagename=None, user=None):
...
+        raise ValidationError(
+            _('Plugin cannot be uninstalled as it is currently active')
+        )
+
+    validate_package_plugin(cfg, user)
+    package_name = cfg.package_name
+    logger.info('Uninstalling plugin: %s', package_name)
+
+    if delete_tables:
+        # TODO: Remove any database tables associated with this plugin
```
Comment (matmair) [inline InvenTree/plugin/installer.py:293]: Question: Will this be part of this PR?

File: InvenTree/InvenTree/api_version.py [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
 # InvenTree API version
-INVENTREE_API_VERSION = 165
+INVENTREE_API_VERSION = 166
 """Increment this API version number whenever there is a significant change to the API that any clients need to know about."""
 
 INVENTREE_API_TEXT = """
 
+v166 -> 2024-02-04 : https://github.com/inventree/InvenTree/pull/6400
+    - Adds package_name to plugin API
+    - Adds mechanism for installing plugins via the API
```
Comment (matmair) [inline InvenTree/InvenTree/api_version.py:11]: Suggestion: mention uninstalling too

File: .vscode/launch.json [hunk 5]
```diff
@@ -10,6 +10,7 @@
       "request": "launch",
       "program": "${workspaceFolder}/InvenTree/manage.py",
       "args": ["runserver"],
+      "pythonArgs": ["-Xfrozen_modules=off"],
```
Comment (matmair) [inline .vscode/launch.json:13]: Question: Why is this added?

**Comment by matmair** [comment]
Looks intresting in general, a few things should probably be their own PR

File: src/frontend/src/components/forms/ApiForm.tsx (hunk 1 above)
Comment (SchrodingersGat) [reply src/frontend/src/components/forms/ApiForm.tsx:300]: Have adjusted this now, as part of `setApiDefaults`

File: src/frontend/src/tables/part/PartTestTemplateTable.tsx (hunk 2 above)
Comment (SchrodingersGat) [reply src/frontend/src/tables/part/PartTestTemplateTable.tsx:88]: Polluted, I will remove

File: InvenTree/plugin/installer.py (hunk 3 above)
Comment (SchrodingersGat) [reply InvenTree/plugin/installer.py:293]: Not sure. Don't think so at this stage - this will be a lot trickier.. I will leave it there as a placeholder and make a new issue

File: InvenTree/InvenTree/api_version.py (hunk 4 above)
Comment (SchrodingersGat) [reply InvenTree/InvenTree/api_version.py:11]: That was just a typo :) Fixed now!

File: .vscode/launch.json (hunk 5 above)
Comment (SchrodingersGat) [reply .vscode/launch.json:13]: Trying to remove an annoying warning in debug console. This did not fix, I will remove it.

File: InvenTree/plugin/installer.py (hunk 3 above)
Comment (SchrodingersGat) [reply InvenTree/plugin/installer.py:293]: Removed for now
